Cannot find module '../common/mvc/class/HandlerBuilder' with version 4.19.0

Hi @Romakita, currently my project starts fine with the latest version `4.19.0` however im not able to run my tests anymore.. it is throwing this error: 

```
/usr/local/bin/node /Users/milewski/Code/yumi-frontend/backend/node_modules/mocha/bin/_mocha test/api/**/*.ts --require ts-node/register --no-timeouts --exit --ui bdd --reporter /Applications/PhpStorm.app/Contents/plugins/NodeJS/js/mocha-intellij/lib/mochaIntellijReporter.js /Users/milewski/Code/yumi-frontend/backend/test
Error: Cannot find module '../common/mvc/class/HandlerBuilder'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:571:15)
    at Function.Module._load (internal/modules/cjs/loader.js:497:25)
    at Module.require (internal/modules/cjs/loader.js:626:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/Users/milewski/Code/yumi-frontend/backend/node_modules/@tsed/src/testing/loadInjector.ts:7:1)
    at Module._compile (internal/modules/cjs/loader.js:678:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:689:10)
    at Module.load (internal/modules/cjs/loader.js:589:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:528:12)
    at Function.Module._load (internal/modules/cjs/loader.js:520:3)
    at Module.require (internal/modules/cjs/loader.js:626:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/Users/milewski/Code/yumi-frontend/backend/node_modules/@tsed/src/testing/inject.ts:2:1)
    at Module._compile (internal/modules/cjs/loader.js:678:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:689:10)
    at Module.load (internal/modules/cjs/loader.js:589:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:528:12)
    at Function.Module._load (internal/modules/cjs/loader.js:520:3)
    at Module.require (internal/modules/cjs/loader.js:626:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/Users/milewski/Code/yumi-frontend/backend/node_modules/@tsed/src/testing/index.ts:1:1)
    at Module._compile (internal/modules/cjs/loader.js:678:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:689:10)
    at Module.load (internal/modules/cjs/loader.js:589:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:528:12)
    at Function.Module._load (internal/modules/cjs/loader.js:520:3)
    at Module.require (internal/modules/cjs/loader.js:626:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/Users/milewski/Code/yumi-frontend/backend/test/api/admin/AdminPlanController.ts:5:19)
    at Module._compile (internal/modules/cjs/loader.js:678:30)
    at Module.m._compile (/Users/milewski/Code/yumi-frontend/backend/node_modules/ts-node/src/index.ts:430:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:689:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/Users/milewski/Code/yumi-frontend/backend/node_modules/ts-node/src/index.ts:433:12)
    at Module.load (internal/modules/cjs/loader.js:589:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:528:12)
    at Function.Module._load (internal/modules/cjs/loader.js:520:3)
    at Module.require (internal/modules/cjs/loader.js:626:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at /Users/milewski/Code/yumi-frontend/backend/node_modules/mocha/lib/mocha.js:250:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/Users/milewski/Code/yumi-frontend/backend/node_modules/mocha/lib/mocha.js:247:14)
    at Mocha.run (/Users/milewski/Code/yumi-frontend/backend/node_modules/mocha/lib/mocha.js:576:10)
    at Object.<anonymous> (/Users/milewski/Code/yumi-frontend/backend/node_modules/mocha/bin/_mocha:637:18)
    at Module._compile (internal/modules/cjs/loader.js:678:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:689:10)
    at Module.load (internal/modules/cjs/loader.js:589:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:528:12)
    at Function.Module._load (internal/modules/cjs/loader.js:520:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:719:10)
    at startup (internal/bootstrap/node.js:228:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:576:3)

Process finished with exit code 1
```

--- a/src/common/filters/class/FilterBuilder.ts
+++ b/src/common/filters/class/FilterBuilder.ts
@@ -11,129 +11,129 @@ import {ValidationService} from "../services/ValidationService";
 import {ParamMetadata} from "./ParamMetadata";
 
 export class FilterBuilder {
-    constructor(private injector: InjectorService) {}
+  constructor(private injector: InjectorService) {}
 
-    /**
-     *
-     */
-    public build(param: ParamMetadata): Function {
-        let filter: any = this.initFilter(param);
-        filter = this.appendRequiredFilter(filter, param);
-        filter = this.appendValidationFilter(filter, param);
-        filter = this.appendConverterFilter(filter, param);
+  /**
+   *
+   */
+  public build(param: ParamMetadata): Function {
+    let filter: any = this.initFilter(param);
+    filter = this.appendRequiredFilter(filter, param);
+    filter = this.appendValidationFilter(filter, param);
+    filter = this.appendConverterFilter(filter, param);
 
-        return filter;
+    return filter;
+  }
+
+  /**
+   *
+   * @param {Type<IFilter>} target
+   * @param args
+   * @returns {any}
+   */
+  private invoke(target: Type<IFilter>, ...args: any[]): any {
+    const instance = this.injector.get<IFilter>(target);
+
+    if (!instance || !instance.transform) {
+      throw new UnknowFilterError(target);
     }
 
-    /**
-     *
-     * @param {Type<IFilter>} target
-     * @param args
-     * @returns {any}
-     */
-    private invoke(target: Type<IFilter>, ...args: any[]): any {
-        const instance = this.injector.get<IFilter>(target);
+    const [expression, request, response] = args;
 
-        if (!instance || !instance.transform) {
-            throw new UnknowFilterError(target);
-        }
+    return instance.transform(expression, request, response);
+  }
 
-        const [expression, request, response] = args;
+  /**
+   *
+   * @param {ParamMetadata} param
+   * @returns {any}
+   */
+  private initFilter(param: ParamMetadata): IFilterPreHandler {
+    if (typeof param.service === "symbol") {
+      const sym = param.service as symbol;
 
-        return instance.transform(expression, request, response);
+      if (FilterPreHandlers.has(sym)) {
+        return FilterPreHandlers.get(sym)!;
+      }
     }
 
-    /**
-     *
-     * @param {ParamMetadata} param
-     * @returns {any}
-     */
-    private initFilter(param: ParamMetadata): IFilterPreHandler {
-        if (typeof param.service === "symbol") {
-            const sym = param.service as symbol;
+    // wrap Custom Filter to FilterPreHandler
+    return (locals: IFilterScope) => {
+      return this.invoke(param.service as Type<any>, param.expression, locals.request, locals.response);
+    };
+  }
 
-            if (FilterPreHandlers.has(sym)) {
-                return FilterPreHandlers.get(sym)!;
-            }
-        }
-
-        // wrap Custom Filter to FilterPreHandler
-        return (locals: IFilterScope) => {
-            return this.invoke(param.service as Type<any>, param.expression, locals.request, locals.response);
-        };
+  /**
+   *
+   * @param filter
+   * @param {ParamMetadata} param
+   * @returns {(value: any) => any}
+   */
+  private appendRequiredFilter(filter: any, param: ParamMetadata): Function {
+    if (!param.required) {
+      return filter;
     }
 
-    /**
-     *
-     * @param filter
-     * @param {ParamMetadata} param
-     * @returns {(value: any) => any}
-     */
-    private appendRequiredFilter(filter: any, param: ParamMetadata): Function {
-        if (!param.required) {
-            return filter;
-        }
+    return FilterBuilder.pipe(filter, (value: any) => {
+      if (param.isRequired(value)) {
+        throw new RequiredParamError(param.name, param.expression);
+      }
 
-        return FilterBuilder.pipe(filter, (value: any) => {
-            if (param.isRequired(value)) {
-                throw new RequiredParamError(param.name, param.expression);
-            }
+      return value;
+    });
+  }
 
-            return value;
-        });
+  /**
+   *
+   * @param filter
+   * @param param
+   * @returns {(value: any) => any}
+   */
+  private appendConverterFilter(filter: any, param: ParamMetadata): Function {
+    if (!param.useConverter) {
+      return filter;
     }
 
-    /**
-     *
-     * @param filter
-     * @param param
-     * @returns {(value: any) => any}
-     */
-    private appendConverterFilter(filter: any, param: ParamMetadata): Function {
-        if (!param.useConverter) {
-            return filter;
-        }
+    const converterService = this.injector.get<ConverterService>(ConverterService)!;
 
-        const converterService = this.injector.get<ConverterService>(ConverterService)!;
+    return FilterBuilder.pipe(filter, converterService.deserialize.bind(converterService), param.collectionType || param.type, param.type);
+  }
 
-        return FilterBuilder.pipe(filter, converterService.deserialize.bind(converterService), param.collectionType || param.type, param.type);
+  /**
+   *
+   * @param filter
+   * @param param
+   * @returns {(value: any) => any}
+   */
+  private appendValidationFilter(filter: any, param: ParamMetadata): Function {
+    const type = param.type || param.collectionType;
+    const {collectionType} = param;
+
+    if (!param.useValidation || (param.useValidation && !type)) {
+      return filter;
     }
 
-    /**
-     *
-     * @param filter
-     * @param param
-     * @returns {(value: any) => any}
-     */
-    private appendValidationFilter(filter: any, param: ParamMetadata): Function {
-        const type = param.type || param.collectionType;
-        const {collectionType} = param;
+    const validationService = this.injector.get<ValidationService>(ValidationService)!;
 
-        if (!param.useValidation || (param.useValidation && !type)) {
-            return filter;
-        }
+    return FilterBuilder.pipe(filter, (value: any) => {
+      try {
+        validationService.validate(value, type, collectionType);
+      } catch (err) {
+        throw new ParseExpressionError(param.name, param.expression, err.message);
+      }
 
-        const validationService = this.injector.get<ValidationService>(ValidationService)!;
+      return value;
+    });
+  }
 
-        return FilterBuilder.pipe(filter, (value: any) => {
-            try {
-                validationService.validate(value, type, collectionType);
-            } catch (err) {
-                throw new ParseExpressionError(param.name, param.expression, err.message);
-            }
-
-            return value;
-        });
-    }
-
-    /**
-     *
-     * @param {Function} filter
-     * @param {Function} newFilter
-     * @param args
-     * @returns {(value: any) => any}
-     */
-    private static pipe(filter: Function, newFilter: Function, ...args: any[]): Function {
-        return (value: any) => newFilter(filter(value), ...args);
-    }
+  /**
+   *
+   * @param {Function} filter
+   * @param {Function} newFilter
+   * @param args
+   * @returns {(value: any) => any}
+   */
+  private static pipe(filter: Function, newFilter: Function, ...args: any[]): Function {
+    return (value: any) => newFilter(filter(value), ...args);
+  }
 }

--- a/src/common/filters/class/FilterBuilder.ts
+++ b/src/common/filters/class/FilterBuilder.ts
@@ -1,7 +1,6 @@
 import {Type} from "@tsed/core";
+import {InjectorService, ParseExpressionError} from "@tsed/common";
 import {ConverterService} from "../../converters/services/ConverterService";
-import {InjectorService} from "../../di/services/InjectorService";
-import {ParseExpressionError} from "../../mvc/errors/ParseExpressionError";
 import {RequiredParamError} from "../errors/RequiredParamError";
 import {UnknowFilterError} from "../errors/UnknowFilterError";
 import {IFilter} from "../interfaces";
@@ -12,129 +11,129 @@ import {ValidationService} from "../services/ValidationService";
 import {ParamMetadata} from "./ParamMetadata";
 
 export class FilterBuilder {
-  constructor(private injector: InjectorService) {}
+    constructor(private injector: InjectorService) {}
 
-  /**
-   *
-   */
-  public build(param: ParamMetadata): Function {
-    let filter: any = this.initFilter(param);
-    filter = this.appendRequiredFilter(filter, param);
-    filter = this.appendValidationFilter(filter, param);
-    filter = this.appendConverterFilter(filter, param);
+    /**
+     *
+     */
+    public build(param: ParamMetadata): Function {
+        let filter: any = this.initFilter(param);
+        filter = this.appendRequiredFilter(filter, param);
+        filter = this.appendValidationFilter(filter, param);
+        filter = this.appendConverterFilter(filter, param);
 
-    return filter;
-  }
-
-  /**
-   *
-   * @param {Type<IFilter>} target
-   * @param args
-   * @returns {any}
-   */
-  private invoke(target: Type<IFilter>, ...args: any[]): any {
-    const instance = this.injector.get<IFilter>(target);
-
-    if (!instance || !instance.transform) {
-      throw new UnknowFilterError(target);
+        return filter;
     }
 
-    const [expression, request, response] = args;
+    /**
+     *
+     * @param {Type<IFilter>} target
+     * @param args
+     * @returns {any}
+     */
+    private invoke(target: Type<IFilter>, ...args: any[]): any {
+        const instance = this.injector.get<IFilter>(target);
 
-    return instance.transform(expression, request, response);
-  }
+        if (!instance || !instance.transform) {
+            throw new UnknowFilterError(target);
+        }
 
-  /**
-   *
-   * @param {ParamMetadata} param
-   * @returns {any}
-   */
-  private initFilter(param: ParamMetadata): IFilterPreHandler {
-    if (typeof param.service === "symbol") {
-      const sym = param.service as symbol;
+        const [expression, request, response] = args;
 
-      if (FilterPreHandlers.has(sym)) {
-        return FilterPreHandlers.get(sym)!;
-      }
+        return instance.transform(expression, request, response);
     }
 
-    // wrap Custom Filter to FilterPreHandler
-    return (locals: IFilterScope) => {
-      return this.invoke(param.service as Type<any>, param.expression, locals.request, locals.response);
-    };
-  }
+    /**
+     *
+     * @param {ParamMetadata} param
+     * @returns {any}
+     */
+    private initFilter(param: ParamMetadata): IFilterPreHandler {
+        if (typeof param.service === "symbol") {
+            const sym = param.service as symbol;
 
-  /**
-   *
-   * @param filter
-   * @param {ParamMetadata} param
-   * @returns {(value: any) => any}
-   */
-  private appendRequiredFilter(filter: any, param: ParamMetadata): Function {
-    if (!param.required) {
-      return filter;
+            if (FilterPreHandlers.has(sym)) {
+                return FilterPreHandlers.get(sym)!;
+            }
+        }
+
+        // wrap Custom Filter to FilterPreHandler
+        return (locals: IFilterScope) => {
+            return this.invoke(param.service as Type<any>, param.expression, locals.request, locals.response);
+        };
     }
 
-    return FilterBuilder.pipe(filter, (value: any) => {
-      if (param.isRequired(value)) {
-        throw new RequiredParamError(param.name, param.expression);
-      }
+    /**
+     *
+     * @param filter
+     * @param {ParamMetadata} param
+     * @returns {(value: any) => any}
+     */
+    private appendRequiredFilter(filter: any, param: ParamMetadata): Function {
+        if (!param.required) {
+            return filter;
+        }
 
-      return value;
-    });
-  }
+        return FilterBuilder.pipe(filter, (value: any) => {
+            if (param.isRequired(value)) {
+                throw new RequiredParamError(param.name, param.expression);
+            }
 
-  /**
-   *
-   * @param filter
-   * @param param
-   * @returns {(value: any) => any}
-   */
-  private appendConverterFilter(filter: any, param: ParamMetadata): Function {
-    if (!param.useConverter) {
-      return filter;
+            return value;
+        });
     }
 
-    const converterService = this.injector.get<ConverterService>(ConverterService)!;
+    /**
+     *
+     * @param filter
+     * @param param
+     * @returns {(value: any) => any}
+     */
+    private appendConverterFilter(filter: any, param: ParamMetadata): Function {
+        if (!param.useConverter) {
+            return filter;
+        }
 
-    return FilterBuilder.pipe(filter, converterService.deserialize.bind(converterService), param.collectionType || param.type, param.type);
-  }
+        const converterService = this.injector.get<ConverterService>(ConverterService)!;
 
-  /**
-   *
-   * @param filter
-   * @param param
-   * @returns {(value: any) => any}
-   */
-  private appendValidationFilter(filter: any, param: ParamMetadata): Function {
-    const type = param.type || param.collectionType;
-    const {collectionType} = param;
-
-    if (!param.useValidation || (param.useValidation && !type)) {
-      return filter;
+        return FilterBuilder.pipe(filter, converterService.deserialize.bind(converterService), param.collectionType || param.type, param.type);
     }
 
-    const validationService = this.injector.get<ValidationService>(ValidationService)!;
+    /**
+     *
+     * @param filter
+     * @param param
+     * @returns {(value: any) => any}
+     */
+    private appendValidationFilter(filter: any, param: ParamMetadata): Function {
+        const type = param.type || param.collectionType;
+        const {collectionType} = param;
 
-    return FilterBuilder.pipe(filter, (value: any) => {
-      try {
-        validationService.validate(value, type, collectionType);
-      } catch (err) {
-        throw new ParseExpressionError(param.name, param.expression, err.message);
-      }
+        if (!param.useValidation || (param.useValidation && !type)) {
+            return filter;
+        }
 
-      return value;
-    });
-  }
+        const validationService = this.injector.get<ValidationService>(ValidationService)!;
 
-  /**
-   *
-   * @param {Function} filter
-   * @param {Function} newFilter
-   * @param args
-   * @returns {(value: any) => any}
-   */
-  private static pipe(filter: Function, newFilter: Function, ...args: any[]): Function {
-    return (value: any) => newFilter(filter(value), ...args);
-  }
+        return FilterBuilder.pipe(filter, (value: any) => {
+            try {
+                validationService.validate(value, type, collectionType);
+            } catch (err) {
+                throw new ParseExpressionError(param.name, param.expression, err.message);
+            }
+
+            return value;
+        });
+    }
+
+    /**
+     *
+     * @param {Function} filter
+     * @param {Function} newFilter
+     * @param args
+     * @returns {(value: any) => any}
+     */
+    private static pipe(filter: Function, newFilter: Function, ...args: any[]): Function {
+        return (value: any) => newFilter(filter(value), ...args);
+    }
 }

--- a/src/common/mvc/index.ts
+++ b/src/common/mvc/index.ts
@@ -9,8 +9,11 @@ import "./utils/extendsRequest";
 export * from "./interfaces";
 
 // provide
+export * from "./class/ControllerBuilder";
+export * from "./class/HandlerBuilder";
 export * from "./class/ControllerProvider";
 export * from "./class/EndpointMetadata";
+export * from "./class/HandlerMetadata";
 export * from "./class/HandlerMetadata";
 
 // registries
@@ -31,6 +34,14 @@ export * from "./services/MiddlewareService";
 export * from "./services/RouterController";
 export * from "./services/ExpressRouter";
 export * from "./services/RouteService";
+
+// errors
+export * from "./errors/CastError";
+export * from "./errors/CyclicReferenceError";
+export * from "./errors/ParseExpressionError";
+export * from "./errors/TemplateRenderingError";
+export * from "./errors/UnknowControllerError";
+export * from "./errors/UnknowMiddlewareError";
 
 // decorators
 export * from "./decorators";

--- a/src/testing/loadInjector.ts
+++ b/src/testing/loadInjector.ts
@@ -1,4 +1,12 @@
-import {ExpressApplication, HttpsServer, InjectorService, ServerSettingsService, HandlerBuilder, HttpServer, IHTTPSServerOptions} from "@tsed/common";
+import {
+  ExpressApplication,
+  HttpsServer,
+  InjectorService,
+  ServerSettingsService,
+  HandlerBuilder,
+  HttpServer,
+  IHTTPSServerOptions
+} from "@tsed/common";
 import {Env} from "@tsed/core";
 
 import * as Express from "express";
@@ -10,23 +18,23 @@ import * as Https from "https";
  * @param {InjectorService} injector
  */
 function expressApplication(injector: InjectorService) {
-    const expressApp = Express();
-    const originalUse = expressApp.use;
+  const expressApp = Express();
+  const originalUse = expressApp.use;
 
-    /* istanbul ignore next */
-    expressApp.use = function (...args: any[]) {
-        args = args.map(arg => {
-            if (injector.has(arg)) {
-                arg = HandlerBuilder.from(arg).build(injector!);
-            }
+  /* istanbul ignore next */
+  expressApp.use = function(...args: any[]) {
+    args = args.map(arg => {
+      if (injector.has(arg)) {
+        arg = HandlerBuilder.from(arg).build(injector!);
+      }
 
-            return arg;
-        });
+      return arg;
+    });
 
-        return originalUse.call(this, ...args);
-    };
+    return originalUse.call(this, ...args);
+  };
 
-    injector!.forkProvider(ExpressApplication, expressApp);
+  injector!.forkProvider(ExpressApplication, expressApp);
 }
 
 /**
@@ -34,12 +42,12 @@ function expressApplication(injector: InjectorService) {
  * @returns {ServerLoader}
  */
 function createHttpServer(injector: InjectorService, port: string | number) {
-    const httpServer: any = Http.createServer(injector!.get(ExpressApplication));
-    // TODO to be removed
-    /* istanbul ignore next */
-    httpServer.get = () => httpServer;
+  const httpServer: any = Http.createServer(injector!.get(ExpressApplication));
+  // TODO to be removed
+  /* istanbul ignore next */
+  httpServer.get = () => httpServer;
 
-    injector!.forkProvider(HttpServer, httpServer);
+  injector!.forkProvider(HttpServer, httpServer);
 }
 
 /**
@@ -60,40 +68,40 @@ function createHttpServer(injector: InjectorService, port: string | number) {
  * @returns {ServerLoader}
  */
 function createHttpsServer(injector: InjectorService, options: IHTTPSServerOptions) {
-    const httpsServer: any = Https.createServer(options, injector!.get(ExpressApplication));
+  const httpsServer: any = Https.createServer(options, injector!.get(ExpressApplication));
 
-    // TODO to be removed
-    /* istanbul ignore next */
-    httpsServer.get = () => httpsServer;
+  // TODO to be removed
+  /* istanbul ignore next */
+  httpsServer.get = () => httpsServer;
 
-    injector!.forkProvider(HttpsServer, httpsServer);
+  injector!.forkProvider(HttpsServer, httpsServer);
 }
 
 export function loadInjector() {
-    const injector = new InjectorService();
-    const hasExpress = injector.has(ExpressApplication);
+  const injector = new InjectorService();
+  const hasExpress = injector.has(ExpressApplication);
 
-    if (!hasExpress) {
-        expressApplication(injector);
-    }
+  if (!hasExpress) {
+    expressApplication(injector);
+  }
 
-    if (!injector.has(HttpServer)) {
-        createHttpServer(injector, 8080);
-    }
+  if (!injector.has(HttpServer)) {
+    createHttpServer(injector, 8080);
+  }
 
-    if (!injector.has(HttpsServer)) {
-        createHttpsServer(injector, {port: 8081});
-    }
+  if (!injector.has(HttpsServer)) {
+    createHttpsServer(injector, {port: 8081});
+  }
 
-    if (!hasExpress) {
-        injector.get<ServerSettingsService>(ServerSettingsService)!.env = Env.TEST;
+  if (!hasExpress) {
+    injector.get<ServerSettingsService>(ServerSettingsService)!.env = Env.TEST;
 
-        /* istanbul ignore next */
-        injector.load().catch(err => {
-            console.error(err);
-            process.exit(-1);
-        });
-    }
+    /* istanbul ignore next */
+    injector.load().catch(err => {
+      console.error(err);
+      process.exit(-1);
+    });
+  }
 
-    return injector;
+  return injector;
 }

--- a/src/testing/loadInjector.ts
+++ b/src/testing/loadInjector.ts
@@ -1,35 +1,32 @@
-import {ExpressApplication, HttpsServer, InjectorService, ServerSettingsService} from "@tsed/common";
+import {ExpressApplication, HttpsServer, InjectorService, ServerSettingsService, HandlerBuilder, HttpServer, IHTTPSServerOptions} from "@tsed/common";
 import {Env} from "@tsed/core";
 
 import * as Express from "express";
 import * as Http from "http";
 import * as Https from "https";
-import {HandlerBuilder} from "../common/mvc/class/HandlerBuilder";
-import {HttpServer} from "../common/server/decorators/httpServer";
-import {IHTTPSServerOptions} from "../common/server/interfaces";
 
 /**
  *
  * @param {InjectorService} injector
  */
 function expressApplication(injector: InjectorService) {
-  const expressApp = Express();
-  const originalUse = expressApp.use;
+    const expressApp = Express();
+    const originalUse = expressApp.use;
 
-  /* istanbul ignore next */
-  expressApp.use = function (...args: any[]) {
-    args = args.map(arg => {
-      if (injector.has(arg)) {
-        arg = HandlerBuilder.from(arg).build(injector!);
-      }
+    /* istanbul ignore next */
+    expressApp.use = function (...args: any[]) {
+        args = args.map(arg => {
+            if (injector.has(arg)) {
+                arg = HandlerBuilder.from(arg).build(injector!);
+            }
 
-      return arg;
-    });
+            return arg;
+        });
 
-    return originalUse.call(this, ...args);
-  };
+        return originalUse.call(this, ...args);
+    };
 
-  injector!.forkProvider(ExpressApplication, expressApp);
+    injector!.forkProvider(ExpressApplication, expressApp);
 }
 
 /**
@@ -37,12 +34,12 @@ function expressApplication(injector: InjectorService) {
  * @returns {ServerLoader}
  */
 function createHttpServer(injector: InjectorService, port: string | number) {
-  const httpServer: any = Http.createServer(injector!.get(ExpressApplication));
-  // TODO to be removed
-  /* istanbul ignore next */
-  httpServer.get = () => httpServer;
+    const httpServer: any = Http.createServer(injector!.get(ExpressApplication));
+    // TODO to be removed
+    /* istanbul ignore next */
+    httpServer.get = () => httpServer;
 
-  injector!.forkProvider(HttpServer, httpServer);
+    injector!.forkProvider(HttpServer, httpServer);
 }
 
 /**
@@ -63,40 +60,40 @@ function createHttpServer(injector: InjectorService, port: string | number) {
  * @returns {ServerLoader}
  */
 function createHttpsServer(injector: InjectorService, options: IHTTPSServerOptions) {
-  const httpsServer: any = Https.createServer(options, injector!.get(ExpressApplication));
+    const httpsServer: any = Https.createServer(options, injector!.get(ExpressApplication));
 
-  // TODO to be removed
-  /* istanbul ignore next */
-  httpsServer.get = () => httpsServer;
+    // TODO to be removed
+    /* istanbul ignore next */
+    httpsServer.get = () => httpsServer;
 
-  injector!.forkProvider(HttpsServer, httpsServer);
+    injector!.forkProvider(HttpsServer, httpsServer);
 }
 
 export function loadInjector() {
-  const injector = new InjectorService();
-  const hasExpress = injector.has(ExpressApplication);
+    const injector = new InjectorService();
+    const hasExpress = injector.has(ExpressApplication);
 
-  if (!hasExpress) {
-    expressApplication(injector);
-  }
+    if (!hasExpress) {
+        expressApplication(injector);
+    }
 
-  if (!injector.has(HttpServer)) {
-    createHttpServer(injector, 8080);
-  }
+    if (!injector.has(HttpServer)) {
+        createHttpServer(injector, 8080);
+    }
 
-  if (!injector.has(HttpsServer)) {
-    createHttpsServer(injector, {port: 8081});
-  }
+    if (!injector.has(HttpsServer)) {
+        createHttpsServer(injector, {port: 8081});
+    }
 
-  if (!hasExpress) {
-    injector.get<ServerSettingsService>(ServerSettingsService)!.env = Env.TEST;
+    if (!hasExpress) {
+        injector.get<ServerSettingsService>(ServerSettingsService)!.env = Env.TEST;
 
-    /* istanbul ignore next */
-    injector.load().catch(err => {
-      console.error(err);
-      process.exit(-1);
-    });
-  }
+        /* istanbul ignore next */
+        injector.load().catch(err => {
+            console.error(err);
+            process.exit(-1);
+        });
+    }
 
-  return injector;
+    return injector;
 }

